packages/flutter_tools/lib/src/base/user_messages.dart

```diff
@@ -292,4 +293,16 @@ class UserMessages {
       'In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.\n'
       'Read more about iOS versioning at\n'
       'https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html\n';
+
+  String get _androidSdkInstallUrl {
```

Comment:
Try:
```dart
import 'globals.dart' as globals;
```

```dart
  String get _androidSdkInstallUrl {
    if (globals.platform.isMacOS) {
      return 'https://flutter.dev/docs/get-started/install/macos#android-setup';
    }
...
```

Reply:
That I cannot compile on my machine 
```
Unable to spawn isolate: lib/src/base/user_messages.dart:6:8: Error: Error when reading 'lib/src/base/globals.dart': No such file or directory
import 'globals.dart' as globals;
       ^
```

Why dart:io is forbidden? Isn't it a standard library of the language?

Reply:
> That I cannot compile on my machine

I was just indicating how this is done everywhere else (so you can mock out the platform you're on).  grep around for `isMacOS` and you'll see how it's done.
You need to adjust the import to point to the relative path:
```
import '../globals.dart' as globals;
```

> Why dart:io is forbidden? Isn't it a standard library of the language?

You would import `lib/src/base/io.dart` if you need something in there, see https://github.com/flutter/flutter/pull/7390 and https://github.com/flutter/flutter/pull/7385 re:testability.
But in your case the mockable platform logic is in `globals.dart` so just import that.

Reply:
Thank you very much for clarification. Done.